RecoTauTag/HLTProducers/src/L1THLTTauMatching.cc

```diff
@@ -0,0 +1,108 @@
+#include "RecoTauTag/HLTProducers/interface/L1THLTTauMatching.h"
+#include "Math/GenVector/VectorUtil.h"
+#include "DataFormats/L1Trigger/interface/Jet.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "DataFormats/TauReco/interface/PFTau.h"
+
+//
+// class decleration
+//
+using namespace reco   ;
+using namespace std    ;
+using namespace edm    ;
+using namespace trigger;
+
+L1THLTTauMatching::L1THLTTauMatching(const edm::ParameterSet& iConfig):
+  jetSrc    ( consumes<PFTauCollection>                     (iConfig.getParameter<InputTag>("JetSrc"      ) ) ),
+  tauTrigger( consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.getParameter<InputTag>("L1TauTrigger") ) ),
+  mEt_Min   (                                                iConfig.getParameter<double>  ("EtMin"       )   )
+{  
+  produces<PFTauCollection>();
+}
+L1THLTTauMatching::~L1THLTTauMatching(){ }
+
+void L1THLTTauMatching::produce(edm::StreamID iSId, edm::Event& iEvent, const edm::EventSetup& iES) const
+{
+	
+  auto_ptr<PFTauCollection> tauL2jets(new PFTauCollection);
+	
+  double deltaR    = 1.0;
+  double matchingR = 0.5;
+  //Getting HLT jets to be matched
+  edm::Handle<PFTauCollection > tauJets;
+  iEvent.getByToken( jetSrc, tauJets );
+
+  //		std::cout <<"Size of input jet collection "<<tauJets->size()<<std::endl;
+		
+  edm::Handle<trigger::TriggerFilterObjectWithRefs> l1TriggeredTaus;
+  iEvent.getByToken(tauTrigger,l1TriggeredTaus);
+				
+  l1t::JetVectorRef tauCandRefVec;
+  l1t::JetVectorRef jetCandRefVec;  
+  l1TriggeredTaus->getObjects( trigger::TriggerL1TauJet,tauCandRefVec);
+  l1TriggeredTaus->getObjects( trigger::TriggerL1CenJet,jetCandRefVec);
```

Comment:
This is deprecated, cen and for jets are combined in one; using trigger::TriggerL1Jet,